src/averageAges.js

```diff
@@ -20,6 +20,15 @@ function calculateMenAverageAge(people, century) {
   // avoid using loop and forEach
   // replace `if ()` statement with &&, || or ?:
   // without nesting
+  const menAges = people.filter((person) => {
+    const deathCentury = Math.ceil(person.died / 100);
+
+    return century !== undefined
+      ? deathCentury === century && person.sex === 'm'
+      : person.sex === 'm';
+  }).map(person => person.died - person.born);
+
+  return menAges.reduce((sum, age) => sum + age, 0) / menAges.length;
```

Comment:
Instead of iterate array twice, u can move logic from map method to reduce. Also please move this condition to separate function, because in calculateWomenAverageAge u use the same
```suggestion
  }).map(person => person.died - person.born);

  return menAges.reduce((sum, person) => sum + (person.died - person.born), 0) / menAges.length;
```
![image](https://github.com/mate-academy/js_average-ages/assets/54936044/7561314f-b4bc-4741-a76f-14d90fef923c)